src/test/java/cloud/fogbow/ras/core/plugins/interoperability/openstack/volume/v2/OpenStackVolumePluginTest.java

```diff
@@ -1,201 +1,524 @@
+        boolean status = this.plugin.isReady(cloudState);
 
         // verify
-        Mockito.verify(this.client).doPostRequest(Mockito.anyString(), Mockito.any(), Mockito.any(OpenStackV3User.class)
-        );
-        Assert.assertEquals(FAKE_VOLUME_ID, instanceString);
+        Assert.assertTrue(status);
+    }
+    
+    // test case: When calling the isReady method with the cloud states different
+    // than available, this means that the state of volume is not READY and it must
+    // return false.
+    @Test
+    public void testIsReadyUnsuccessful() {
+        // set up
+        String[] cloudStates = { OpenStackStateMapper.CREATING_STATUS, OpenStackStateMapper.ERROR_STATUS,
+                OpenStackStateMapper.IN_USE_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.isReady(cloudState);
+
+            // verify
+            Assert.assertFalse(status);
+        }
+    }
+    
+    // test case: When calling the hasFailed method with any cloud states errors,
+    // this means that the state of the volume is FAILED and it must return true.
+    @Test
+    public void testHasFailedSuccessful() {
+        // set up
+        String[] cloudStates = { OpenStackStateMapper.ERROR_BACKING_UP_STATUS,
+                OpenStackStateMapper.ERROR_DELETING_STATUS, OpenStackStateMapper.ERROR_EXTENDING_STATUS,
+                OpenStackStateMapper.ERROR_RESTORING_STATUS, OpenStackStateMapper.ERROR_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.hasFailed(cloudState);
+
+            // verify
+            Assert.assertTrue(status);
+        }
     }
+    
+    // test case: When calling the hasFailed method with the cloud states different
+    // than any error, this means that the state of the volume is not FAILED and it
+    // must return false.
+    @Test
+    public void testhasFailedUnsuccessful() {
+        // set up
+        String[] cloudStates = { OpenStackStateMapper.CREATING_STATUS, OpenStackStateMapper.AVAILABLE_STATUS,
+                OpenStackStateMapper.IN_USE_STATUS };
+
+        for (String cloudState : cloudStates) {
+            // exercise
+            boolean status = this.plugin.hasFailed(cloudState);
 
-    // test case: Check if the request in requestInstance() is executed properly with the right parameters even when
-    // there is volume extra requirements.
+            // verify
+            Assert.assertFalse(status);
+        }
+    }
+    
+    // test case: When invoking the requestInstance method with a valid volume
+    // request and a cloud user, it must verify that the call was successful.
     @Test
-    public void testRequestInstanceWithRequirements() throws FogbowException, HttpResponseException {
+    public void testRequestInstance() throws FogbowException {
```

Comment:
you should verify if findVolumeTypeId has been called. Same for defineInstanceName.

Reply:
Okay, I will add these checks.